### PR TITLE
Validations and tests on my models

### PR DESCRIPTION
**Pull request by sabetech**
In this pull request
- Implemented Rails based Validation on my models
- Wrote tests for the models
  - User
  - Like
  - Comment
  - Post

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-      user.posts_count += 1
-    end
+  after_save :update_post_counter
 
-    user.save
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_count, numericality: { only_integer: true }
+  validates :comments_count, comparison: { greater_than_or_equal_to: 0 }
+  validates :comments_count, numericality: { only_integer: true }
+  validates :comments_count, comparison: { greater_than_or_equal_to: 0 }
```
Comment (Deepakdanger) [inline app/models/post.rb]: - [ ] Here, `comments_count` got validation again, I think here we should validate `likes_count` here also.

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class User < ApplicationRecord
   has_many :comments, foreign_key: 'user_id'
   has_many :likes, foreign_key: 'user_id'
 
+  validates :name, presence: true
```
Comment (Deepakdanger) [inline app/models/user.rb:6]: - [ ] [OPTIONAL] Here if we put the validation length of the `name` 3 or more than 3, then it will be better validation.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @sabetech , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Kindly check all the test cases again before submission.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @sabetech,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**